File: app/src/main/java/com/denchic45/kts/uivalidator/validatorlistener/ValidationListener.java
package com.denchic45.kts.uivalidator.validatorlistener;

import com.denchic45.kts.uivalidator.Rule;

public abstract class ValidationListener {

    public static final ValidationListener EMPTY = new ValidationListener() {
        @Override
        void onSuccess() {
        }

        @Override
        void onError(Rule rule) {
        }
    };

    public final void run(Rule rule) {
        if (rule.isValid())
            onSuccess();
        else
            onError(rule);
    }

    abstract void onSuccess();

    abstract void onError(Rule rule);
}

File: android/src/main/java/com/denchic45/kts/uivalidator/validatorlistener/ValidationListener.kt
package com.denchic45.kts.uivalidator.validatorlistener

import com.denchic45.kts.uivalidator.Rule

abstract class ValidationListener {
    fun run(rule: Rule) {
        if (rule.isValid) onSuccess() else onError(rule)
    }

    abstract fun onSuccess()
    abstract fun onError(rule: Rule)

    companion object {
        val EMPTY: ValidationListener = object : ValidationListener() {
            override fun onSuccess() {}
            override fun onError(rule: Rule) {}
        }
    }
}